### PR TITLE
fix(data): tests for error response fix

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/api-graphql/__tests__/internals/generateClient.test.ts [hunk 1]
```diff
@@ -645,7 +644,7 @@ describe('generateClient', () => {
 				},
 			});
 
-			const { data: notes } = await data.notes();
+			const { data: notes } = await data!.notes();
```
Comment (HuiSF) [inline packages/api-graphql/__tests__/internals/generateClient.test.ts:647]: nit: I'd assert `expect(data).not.toBeNull()` before force unwrap it as it's also an expected behavior?

File: packages/api-graphql/__tests__/internals/generateClient.test.ts [hunk 2]
```diff
@@ -835,6 +834,186 @@ describe('generateClient', () => {
...
+							line: 2,
+							column: 3,
+							sourceName: null,
+						},
+					],
+					message: 'Not Authorized to access updateTodo on type Mutation',
+				},
+			];
+
+			const spy = mockApiResponse({
```
Comment (HuiSF) [inline packages/api-graphql/__tests__/internals/generateClient.test.ts:926]: nit: probably don't this this assignment...

File: packages/api-graphql/__tests__/internals/generateClient.test.ts [hunk 3]
```diff
@@ -835,6 +834,186 @@ describe('generateClient', () => {
...
+			const { data, errors } = await client.models.Todo.create({
+				id: 'does not matter',
+			});
+
+			expect(data).toBeNull();
+			expect(errors?.length).toBe(1);
+			expect(errors).toEqual(expectedErrors);
+		});
+
+		test('get() returns null with errors property', async () => {
```
Comment (HuiSF) [inline packages/api-graphql/__tests__/internals/generateClient.test.ts:873]: nit: this test description is not straightforward to understand what's the test without looking the test body... wdyt?

```suggestion
		test('returns object in which `data` is null and `errors` contains expected error', async () => {
```

File: packages/api-graphql/__tests__/internals/generateClient.test.ts [hunk 4]
```diff
@@ -5745,7 +5922,7 @@ describe('generateClient', () => {
 
 			// TODO: data should actually be null/undefined, pending discussion and fix.
 			// This is not strictly related to custom ops.
-			expect(data).toEqual({});
+			expect(data).toEqual(null);
```
Comment (HuiSF) [inline packages/api-graphql/__tests__/internals/generateClient.test.ts:5925]: nit:

```suggestion
			expect(data).toBeNull();
```

**Review by HuiSF — COMMENTED** [review]
Some nitpicking comments 👍

File: packages/api-graphql/__tests__/internals/generateClient.test.ts (hunk 1 above)
Comment (david-mcafee) [reply packages/api-graphql/__tests__/internals/generateClient.test.ts:651]: Good call! I've added this here, as well as everywhere else that I've added the force unwrap in these tests :)

**Review by iartemiev — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Pull request merged**